src/mod/endpoints/mod_sofia/sofia_media.c

```diff
@@ -115,14 +113,14 @@ SWITCH_DECLARE_NONSTD(switch_status_t) switch_string_stream_write(switch_stream_
 }
 
 
-static void process_mp(switch_core_session_t *session, switch_stream_handle_t *stream, const char *boundary, const char *str, const char*isup, intptr_t isup_len, switch_bool_t drop_sdp) {
+static void process_mp(switch_core_session_t *session, switch_stream_handle_t *stream, const char *boundary, const char *str, const char*isup, intptr_t isup_len, switch_bool_t drop_sdp, switch_bool_t allow_sdp) {
 	char *dname = switch_core_session_strdup(session, str);
 	char *dval;
 
 	if ((dval = strchr(dname, ':'))) {
 		*dval++ = '\0';
 		
-		if (drop_sdp && (strcasecmp(dname, "application/sdp") == 0)) {
+		if (drop_sdp && (strcasecmp(dname, "application/sdp") == 0) && !allow_sdp) {
```

Comment:
drop_sdp and not allow_sdp seems to be similar purpose. Why not just use drop_sdp instead. See my other comment for details